osu.Game/Graphics/UserInterfaceV2/DirectorySelector.cs

```diff
@@ -90,6 +92,15 @@ private void updateDisplay(ValueChangedEvent<DirectoryInfo> directory)
                     foreach (var drive in drives)
                         directoryFlow.Add(new DirectoryPiece(drive.RootDirectory));
                 }
+                else if (!directory.NewValue.Exists)
+                {
+                    var newDirectory = directory.NewValue;
+
+                    while (newDirectory != null && !newDirectory.Exists)
```

Comment:
can it ever actually trigger `!Exists`? seems a bit weird.

Reply:
Do you mean the `else if` condition or the one in the condition of the `while` loop?

Reply:
But either way, yes, why wouldn't it?

Reply:
Best to add comments explaining when null or exists are potentially going to cause exit, since it's not immediately obvious.